fjord/base/utils.py

```diff
@@ -323,14 +323,30 @@ def ratelimit(rulename, keyfun=actual_ip, rate='5/m'):
         throttling. defaults to actual_ip.
     :arg rate: (optional) rate to throttle at. defaults to 5/m.
 
+    .. Note::
+
+        Unlike django-ratelimit's ratelimit decorator, this does
+        **not** redirect the user or handle the rate limiting.  Caller
+        must check the request for 'limited' and redirect/handle
+        accordingly.
+
+        For example::
+
+            @ratelimit(rulename='foo', keyfun=actual_ip, rate='1/10m')
+            def view(request):
+                if getattr(request, 'limited', False):
+                    # handle ratelimit here!
+                    raise PermissionDenied()
+
     """
     def decorator(fn):
         @wraps(fn)
         def _wrapped(request, *args, **kwargs):
             already_limited = getattr(request, 'limited', False)
             ratelimited = is_ratelimited(
-                request=request, increment=True, ip=False, method=['POST'],
-                field=None, rate=rate, keys=keyfun)
+                request=request, group=rulename,
+                key=keyfun, rate=rate, method=['POST'],
+                increment=True)
```

Comment:
Documentation for the new `is_ratelimited`: http://django-ratelimit.readthedocs.org/en/latest/usage.html#is_ratelimited

We use the rulename as the group since that's essentially what rulename is. If we didn't pass in rulename as the group value, then it'd generate a group based on the dotted name of the function. That doesn't work here and would create problems.